### PR TITLE
[Feature] Open enrollment administration

### LINKED ISSUE #775: [Feature] Open enrollment administration
- [x] Add
  - [x] "account/benefits/open-enrollment" Route
  - [x] "account/benefits/open-enrollment/add" Route
  - [x] "account/benefits/open-enrollment/edit" Route
- [x] Features
  - [x] View all open enrollment periods
  - [x] Add
  - [x] Edit
  - [x] Delete
  - [x] Suggest from carrier link (AE trust has the same open enrollment policies for all companies)

### PR DESCRIPTION
**Pull request by JavaLavaMT**
closes #775

image

image

image

image

image

### CODE REVIEW

File: app/config.js [hunk 1]
```diff
@@ -66,7 +66,9 @@ const carriers = [{
   name:  'Associated Employers Group Benefit Plan & Trust',
   key:   'ae',
   logo:  'https://granitehr-dev.s3.amazonaws.com/uploads/5d6e7d2fd41c6c27661eb678/5f08c43009c42358fdc6fc3e_83SDcMJFX8szueppRoV3.png',
-  phone: '1 (406) 248-6224'
+  phone: '1 (406) 248-6224',
+  start: '2/17', //get dates
+  end:   '3/12'  //get dates
```
Comment (James1x0) [inline app/config.js:71]: Let's make this an array of `openEnrollments`. For AE it's:
```
Month of december
Month of march
```

File: app/controllers/account/benefits/open-enrollment/add.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import Controller from 'granite/core/controller';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+export const openEnrollmentForm = [{
```
Comment (James1x0) [inline app/controllers/account/benefits/open-enrollment/add.js:4]:
```suggestion

export const openEnrollmentForm = [{
```

File: app/controllers/account/benefits/open-enrollment/add.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  startMode:  'month',
+  path:       'start',
+  formatter:  {
+    header (date) {
+      return moment(date).format('MMM');
+    },
+    date (date) {
+      var day = date.getDate();
+      var month = date.getMonth() + 1;
+      return month + '/' + day;
```
Comment (James1x0) [inline app/controllers/account/benefits/open-enrollment/add.js:17]:
```suggestion
      return (date.getMonth() + 1) + '/' + date.getDate();
```

File: app/controllers/account/benefits/open-enrollment/add.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  type:       'date',
+  startMode:  'month',
+  path:       'start',
+  formatter:  {
+    header (date) {
+      return moment(date).format('MMM');
+    },
+    date (date) {
+      var day = date.getDate();
+      var month = date.getMonth() + 1;
```
Comment (James1x0) [inline app/controllers/account/benefits/open-enrollment/add.js:16]:
```suggestion
```

File: app/controllers/account/benefits/open-enrollment/add.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  startMode:  'month',
+  path:       'end',
+  formatter:  {
+    header (date) {
+      return moment(date).format('MMM');
+    },
+    date (date) {
+      var day = date.getDate();
+      var month = date.getMonth() + 1;
+      return month + '/' + day;
```
Comment (James1x0) [inline app/controllers/account/benefits/open-enrollment/add.js:34]: Same changes as above.

File: app/controllers/account/benefits/open-enrollment/add.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    }
+  }
+}];
+
+export default class AddController extends Controller {
+  @service store
+  @service auth
```
Comment (James1x0) [inline app/controllers/account/benefits/open-enrollment/add.js:41]: Do you really need auth/store?

File: app/routes/account/benefits/open-enrollment/edit.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import Route from 'granite/core/route';
+
+export default class EditRoute extends Route {
+ titleToken = 'edit'
+
+ async model (params) {
+   return  await this.store.find('openEnrollment', params.enrollment_id);
```
Comment (James1x0) [inline app/routes/account/benefits/open-enrollment/edit.js:7]:
```suggestion
   return  await this.store.find('open-enrollment', params.enrollment_id);
```

File: app/serializers/open-enrollment.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
+import classic from 'ember-classic-decorator';
+import ApplicationSerializer from './application';
+import normalizeKeys from '../utils/serialize-object';
+
+@classic
+export default class OpenEnrollment extends ApplicationSerializer {
```
Comment (James1x0) [inline app/serializers/open-enrollment.js:6]: Don't introduce the @classic decorator on new things

File: app/serializers/open-enrollment.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+  normalize (modelClass, hash) {
+    normalizeKeys(hash, true, 'start', 'end');
+    hash.start = `${hash.startMonth}/${hash.startDay}`;
```
Comment (James1x0) [inline app/serializers/open-enrollment.js:9]: Shouldn't this be a date?

File: app/templates/account/benefits/open-enrollment.hbs [hunk 10]
```diff
@@ -0,0 +1,13 @@
+<hgroup class="clearfix">
+  <h1 class="ui header left floated">Open Enrollments</h1>
```
Comment (James1x0) [inline app/templates/account/benefits/open-enrollment.hbs:2]: This header situation is a little off... + icon isn't aligned with Open Enrollments text

File: app/templates/account/benefits/open-enrollment/add.hbs [hunk 11]
```diff
@@ -0,0 +1,16 @@
+<form class="ui form" {{on "submit" (prevent-default (fn this.data.saveRecord this.model 'working' this.afterSaveOptions))}}>
+  <QuickForm class="ui two fields"
+    @form={{this.enrollmentForm}}
+    @model={{this.model}}
+    @controller={{this}}
+    @inputClass="field"
+    @tagName="div" 
+  />
+  <div class="ui divider"></div>
+  <Form::ActionButton type="submit" class="ui massive fluid green button" @disabled={{not (and this.model.end this.model.start)}}>Add</Form::ActionButton>
```
Comment (James1x0) [inline app/templates/account/benefits/open-enrollment/add.hbs:10]: Add the loading attr from data service.

File: app/templates/account/benefits/open-enrollment/edit.hbs [hunk 12]
```diff
@@ -0,0 +1,16 @@
+<form class="ui form" {{on "submit" (prevent-default (fn this.data.saveRecord this.model 'working' this.afterSaveOptions))}}>
+  <QuickForm class="ui two fields"
+    @form={{this.enrollmentForm}}
+    @model={{this.model}}
+    @controller={{this}}
+    @inputClass="field"
+    @tagName="div" 
+  />
+  <div class="ui divider"></div>
+  <Form::ActionButton type="submit" class="ui massive fluid green button" @disabled={{not (and this.model.end this.model.start)}}>Update</Form::ActionButton>
```
Comment (James1x0) [inline app/templates/account/benefits/open-enrollment/edit.hbs:10]: Add the loading attr from data service.

File: app/templates/account/benefits/open-enrollment/index.hbs [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+        </div>
+        <div class="meta">
+          <span class="ui small text right floated">Created {{time-ago enrollment.created}}</span>
+        </div>
+      </div>
+    </div>
+    {{else}}
+      <div class="ui relaxed divided list">
+        <h2 class="ui center aligned header">Nothing to see here.</h2>
+        <h5 class="center aligned text header">There are no Open Enrollments set up.
```
Comment (James1x0) [inline app/templates/account/benefits/open-enrollment/index.hbs:28]:
```suggestion
        <h5 class="center aligned text header">No open enrollments available.
```

File: app/templates/account/benefits/open-enrollment/index.hbs [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+          <LinkTo @route="account.benefits.open-enrollment.add">Add Open Enrollment</LinkTo>
```
Comment (James1x0) [inline app/templates/account/benefits/open-enrollment/index.hbs:29]:
```suggestion
          <LinkTo @route="account.benefits.open-enrollment.add">Add one</LinkTo>
```

File: app/config.js (hunk 1 above)
Comment (JavaLavaMT) [reply app/config.js:71]: 🏁

File: app/controllers/account/benefits/open-enrollment/add.js (hunk 2 above)
Comment (JavaLavaMT) [reply app/controllers/account/benefits/open-enrollment/add.js:4]: 🏁

File: app/controllers/account/benefits/open-enrollment/add.js (hunk 3 above)
Comment (JavaLavaMT) [reply app/controllers/account/benefits/open-enrollment/add.js:17]: 🏁

File: app/controllers/account/benefits/open-enrollment/add.js (hunk 4 above)
Comment (JavaLavaMT) [reply app/controllers/account/benefits/open-enrollment/add.js:16]: 🏁

File: app/controllers/account/benefits/open-enrollment/add.js (hunk 5 above)
Comment (JavaLavaMT) [reply app/controllers/account/benefits/open-enrollment/add.js:34]: 🏁

File: app/templates/account/benefits/open-enrollment/index.hbs (hunk 13 above)
Comment (JavaLavaMT) [reply app/templates/account/benefits/open-enrollment/index.hbs:28]: 🏁

File: app/templates/account/benefits/open-enrollment/index.hbs (hunk 14 above)
Comment (JavaLavaMT) [reply app/templates/account/benefits/open-enrollment/index.hbs:29]: 🏁

File: app/templates/account/benefits/open-enrollment/add.hbs (hunk 11 above)
Comment (JavaLavaMT) [reply app/templates/account/benefits/open-enrollment/add.hbs:10]: 🏁

Comment (JavaLavaMT) [reply app/templates/account/benefits/open-enrollment/add.hbs:10]: 🏁

File: app/templates/account/benefits/open-enrollment.hbs (hunk 10 above)
Comment (JavaLavaMT) [reply app/templates/account/benefits/open-enrollment.hbs:2]: 🏁

File: app/templates/account/benefits/open-enrollment/edit.hbs (hunk 12 above)
Comment (JavaLavaMT) [reply app/templates/account/benefits/open-enrollment/edit.hbs:10]: 🏁

**Comment by JavaLavaMT** [comment]
image

image

File: app/controllers/account/benefits/open-enrollment/add.js (hunk 6 above)
Comment (JavaLavaMT) [reply app/controllers/account/benefits/open-enrollment/add.js:38]: 🏁

File: app/routes/account/benefits/open-enrollment/edit.js (hunk 7 above)
Comment (JavaLavaMT) [reply app/routes/account/benefits/open-enrollment/edit.js:7]: 🏁

File: app/serializers/open-enrollment.js (hunk 8 above)
Comment (JavaLavaMT) [reply app/serializers/open-enrollment.js:6]: 🏁

File: app/serializers/open-enrollment.js (hunk 9 above)
Comment (JavaLavaMT) [reply app/serializers/open-enrollment.js:9]: 🏁

**Pull request merged**